alphagov/paas-release-ci
# [#154957353] Use initial version and initial content for S3 resources instead of s3init.sh

## What

The default S3 resourse doesn't handle an initial version and content and we
had to initialise these files in separate task using s3init.sh.

We already had a fork of the S3 resource to be able to use IAM instance
profiles for authentication [1], so we added new parameters to the resource so
it can emit an initial version with initial content if no versions exist.

We can pass binary content using the initial_content_binary parameter and
passing the base64 encoded value. We do this for tar.gz files where we pass an
empty tar.gz.

[1] https://github.com/alphagov/paas-s3-resource

## How to review

Review https://github.com/alphagov/paas-s3-resource/pull/8 first and update this PR's `[TMP]`commit with the merged tag from DockerHub.

We have successfully tested a build Concourse, but you may want to try it yourself:

1. Deploy a new CI environment from scratch from paas-bootstrap:
  ```
cd paas-bootstrap
BRANCH=remove_s3_init_154957353 make dev build-concourse bootstrap
```
1. Go to localhost:8080 and run the create-bosh-concourse pipeline
1. Upload the pipelines from this branch:
  ```
cd paas-release-ci
CF_DEPLOY_ENV=... DEPLOY_ENV=... BRANCH=remove_s3_init_154957353 SELF_UPDATE_PIPELINE=false make dev pipelines
```
1. Go to https://concourse.${DEPLOY_ENV}.dev.cloudpipeline.digital and run the setup task, it should succeed.
1. Run the `DEPLOY_ENV=[your ci env] make dev upload-cf-cli-secrets` to upload your CF cli creds  
1. Run the rubbernecker pipeline and the paas-billing integration-test pipeline to test a deploy-app pipeline and an integration test pipeline

## Who can review

Not @46bit or me.
